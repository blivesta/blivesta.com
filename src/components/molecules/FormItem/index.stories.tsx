import React from 'react';
import styled from 'styled-components';
import { storiesOf } from '@storybook/react';
import Input from '../../atoms/Input';
import FormItem from '.';

const Wrapper = styled.div`
  width: 480px;
  padding: 60px;
`;

storiesOf(`molecules|FormItem`, module)
  .add(`default`, () => (
    <Wrapper>
      <FormItem title="foo" htmlFor="foo">
        <Input id="foo" placeholder="foo" />
      </FormItem>
    </Wrapper>
  ))
  .add(`required`, () => (
    <Wrapper>
      <FormItem title="bar" htmlFor="bar" required>
        <Input id="bar" aria-required required />
      </FormItem>
    </Wrapper>
  ));
