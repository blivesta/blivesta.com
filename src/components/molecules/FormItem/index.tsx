import * as React from 'react';
import styled from 'styled-components';

import P from '../../atoms/P';

const Label = styled.label`
  font-family: ${props => props.theme.fonts.en};
  letter-spacing: 0.25em;
  text-transform: uppercase;
  text-indent: 0.25em;
  margin-bottom: ${props => props.theme.spaces[1]};
`;

const Item = styled.div`
  flex: 1;
  max-width: 100%;
`;

const Required = styled.span`
  padding-left: ${props => props.theme.spaces[0]};
`;

interface FormItemProps {
  title: string;
  htmlFor?: string;
  children: React.ReactNode;
  required?: boolean;
}

const FormItem = ({ title, required, htmlFor, children }: FormItemProps) => (
  <P>
    <Label htmlFor={htmlFor}>
      {title}
      {required && <Required>*</Required>}
    </Label>
    <Item>{children}</Item>
  </P>
);

export default FormItem;
