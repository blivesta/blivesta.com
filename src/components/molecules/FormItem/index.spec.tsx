import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FormItem from '.';

describe('FormItem', () => {
  it('snapshot', () => {
    const tree = shallow(<FormItem title="title">...</FormItem>);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
