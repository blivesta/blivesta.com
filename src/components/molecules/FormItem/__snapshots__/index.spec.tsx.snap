// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormItem snapshot 1`] = `
<P>
  <styled.label>
    title
  </styled.label>
  <styled.div>
    ...
  </styled.div>
</P>
`;
